utils/sysfs/sysfs.go

```diff
@@ -335,25 +352,205 @@ func (fs *realSysFs) GetSystemUUID() (string, error) {
 	}
 }
 
-func (fs *realSysFs) IsCPUOnline(dir string) bool {
-	cpuPath := fmt.Sprintf("%s/online", dir)
-	content, err := ioutil.ReadFile(cpuPath)
+func (fs *realSysFs) IsCPUOnline(cpuPath string) bool {
+	return isCPUOnline(cpuPath)
+}
+
+// isCPUOnline verifies if a CPU in online. Takes path to /sys/devices/system/cpu/cpu*/ directory as a parameter.
+func isCPUOnline(cpuPath string) bool {
+	if isx86() && isZeroCPU(cpuPath) {
+		// It *is* possible to enable CPU hotplug for CPU0 but it requires setting a boot flag.
+		// It is assumed it's a rare situation.
+		return true
+	}
+
+	const isOnline = 49
+	onlinePath := filepath.Join(cpuPath, "online")
+	onlineVal, err := ioutil.ReadFile(onlinePath)
+	// First, make sure that no error occurred.
 	if err != nil {
-		pathErr, ok := err.(*os.PathError)
-		if ok {
-			if errors.Is(pathErr.Unwrap(), os.ErrNotExist) && isZeroCPU(dir) {
+		// If error occurred and is os.PathError then it might be os.ErrNotExists.
+		pathErr, isPathError := err.(*os.PathError)
```

Comment:
is it needed to do type assertion here? maybe can call os.IsNotExist directly, e.g. https://github.com/google/cadvisor/blob/master/machine/operatingsystem_unix.go#L41-L42

Reply:
Fixed.